samples/apps/copilot-chat-app/webapi/Skills/ChatSkill.cs

```diff
@@ -323,44 +335,80 @@ public async Task<SKContext> ChatAsync(string message, SKContext context)
             return context;
         }
 
-        // Clone the context to avoid modifying the original context variables.
-        var chatContext = Utilities.CopyContextWithVariablesClone(context);
-        chatContext.Variables.Set("knowledgeCutoff", this._promptSettings.KnowledgeCutoffDate);
-        chatContext.Variables.Set("audience", userName);
-
-        // Extract user intent and update remaining token count
-        var userIntent = await this.ExtractUserIntentAsync(chatContext);
-        if (chatContext.ErrorOccurred)
+        if (chatContext.Variables.Get("userCancelledPlan", out var userCancelledPlan))
         {
-            return chatContext;
+            response = "Sorry this plan didn't meet your goals. How can I help you today?";
```

Comment:
Do we want to have a pre-generated response like this or have the bot generate one? This way saves some time in response generation / roundtrips to the kernel

Reply:
Pre-generated will save on costing. Plus since the accept/cancel options are buttons, we arent getting more context from the user to drive a conversation. 
Perhaps reword it to "Sorry this plan didn't meet your goals. How shall I change this plan to better meet your requirements?"